community/lms/doctype/lesson/lesson.py

```diff
@@ -43,3 +43,54 @@ def get_prev(self):
         The return value would be like 1.2, 2.1 etc.
         It will be None if there is no next lesson.
         """
+
+    def get_progress(self):
+        return frappe.db.get_value("LMS Course Progress", {"lesson": self.name, "owner": frappe.session.user}, "status")
+
+    def get_slugified_class(self):
+        if self.get_progress():
+            return ("").join([ s for s in self.get_progress().lower().split() ])
+        return
+
+@frappe.whitelist()
+def save_progress(lesson):
+    if frappe.db.exists("LMS Course Progress",
+            {
+                "lesson": lesson,
+                "owner": frappe.session.user
+            }):
+        return
+
```

Comment:
This API can be called by any user, who is not even a member any batch. I don't think we should allow that. We need to have a check to make sure that won't happen.